src/libraries/System.Reflection.Metadata/src/System/Reflection/Internal/Utilities/BlobUtilities.cs

```diff
@@ -62,60 +36,47 @@ public static void WriteByte(this byte[] buffer, int start, byte value)
 
         public static void WriteUInt16(this byte[] buffer, int start, ushort value)
         {
-            fixed (byte* ptr = &buffer[start])
+            if (!BitConverter.IsLittleEndian)
             {
-                unchecked
-                {
-                    ptr[0] = (byte)value;
-                    ptr[1] = (byte)(value >> 8);
-                }
+                value = BinaryPrimitives.ReverseEndianness(value);
             }
+            Unsafe.WriteUnaligned(ref buffer[start], value);
```

Comment:
Nit: You could tidy this (and the others like it) up to just be:
```C#
public static void WriteUInt16(this byte[] buffer, int start, ushort value) =>
    Unsafe.WriteUnaligned(ref buffer[start], BitConverter.IsLittleEndian ? BinaryPrimitives.ReverseEndianness(value) : value);
```

Reply:
Done.